json-rpc/src/methods.rs

```diff
@@ -103,10 +103,30 @@ async fn get_transactions(
 
     let mut result = vec![];
 
+    let all_events = if include_events {
+        txs.events
+            .ok_or_else(|| format_err!("Storage layer didn't return events when requested!"))?
+    } else {
+        vec![]
+    };
+
     for (v, tx) in txs.transactions.into_iter().enumerate() {
+        let events = if include_events {
+            all_events
+                .get(v)
+                .ok_or_else(|| format_err!("Missing events for version: {}", v))?
+                .iter()
```

Comment:
I may be wrong, but we might be able to use `into_iter` to combine `iter` and `cloned`?

Reply:
no:   Yeah i thought ` all_events.get() gives you &Vec<ContractEvent> , which  I should be able to into_iter() and map() , but turns out doesn't matter &Vec<>.iter() or .into_iter()  gives you the same thing : &ContractEvent